common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -918,4 +967,8 @@ public static String getClosestSuggestionsOnInvalidName(
 
     return String.join(", ", suggestions);
   }
+
+  public static List<Collation> fetchAllCollations() {
```

Comment:
This operation seems a bit too expensive. We always build the whole table and then do a like on it. We need to do something similar to show tables where we have the pattern, as we do not want to ask for collation information if like operator is not concerned with it. Also it feels pretty weird to have `fetchAllCollations` in many places like this. This is not as expensive as building the whole table, but we are also allocating ArrayList multiple times

Reply:
Let me try to think of a better way.